src/app/dashboard/dashboard.component.html

```diff
@@ -44,6 +44,7 @@ <h1 aligan="center">Dashboard</h1>
         {{ profile.username }}
         <button (click)="logout()">Logout</button>
       </div>
+      <h1><button (click)="tableOfProduct()">Table of products</button></h1>
```

Comment:
тут click ивент вообще лишний, тебе надо ссылка и routerLink
```suggestion
      <a routerLink="/products"><button>Table of products</button></a>
```